### PR TITLE
SR parameterized test, BF16 conv fix

### PR DESCRIPTION
**Pull request by nkmellem**

### CODE REVIEW

File: samples/eltwise/run_eltwise_avx512.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (alheinecke) [inline samples/eltwise/run_eltwise_avx512.sh:1]: please remove the entire script

File: samples/eltwise/eltwise_common.h [hunk 2]
```diff
@@ -14,6 +14,118 @@
 #include <math.h>
 #include <string.h>
```
Comment (alheinecke) [inline samples/eltwise/eltwise_common.h:16]: can you please remove this code

File: include/libxsmm_typedefs.h [hunk 3]
```diff
@@ -337,7 +337,7 @@ typedef enum libxsmm_meltw_unary_type {
   LIBXSMM_MELTW_TYPE_UNARY_DECOMPRESS_SPARSE_FACTOR_16  = 49,
   LIBXSMM_MELTW_TYPE_UNARY_DECOMPRESS_SPARSE_FACTOR_32  = 50,
   LIBXSMM_MELTW_TYPE_UNARY_GATHER                       = 51,
-  LIBXSMM_MELTW_TYPE_UNARY_SCATTER                      = 52
+  LIBXSMM_MELTW_TYPE_UNARY_SCATTER                      = 52,
```
Comment (alheinecke) [inline include/libxsmm_typedefs.h:340]: can you please remove the comma

File: samples/eltwise/run_test_avx512.sh [hunk 4]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
```
Comment (alheinecke) [inline samples/eltwise/run_test_avx512.sh:1]: why this new script? For full support (=avx512) we have run_test.sh, we just have AVX2/SSE/aarch64 scripts as they don't implement all TPPs right now. Therefore I suggest removing this script.

File: src/generator_common_x86.c [hunk 5]
```diff
@@ -1960,21 +1960,23 @@ void libxsmm_generator_vcvtneps2bf8_generic_avx512_prep_stack( libxsmm_generated
...
 
-
 LIBXSMM_API_INTERN
 void libxsmm_generator_vcvtneps2bf16_avx512_clean_stack( libxsmm_generated_code* io_generated_code,
                                                          const unsigned int      io_gp_reg ) {
   libxsmm_x86_instruction_pop_reg( io_generated_code, io_gp_reg );
   libxsmm_x86_instruction_pop_reg( io_generated_code, io_gp_reg );
   libxsmm_x86_instruction_pop_reg( io_generated_code, io_gp_reg );
   libxsmm_x86_instruction_pop_reg( io_generated_code, io_gp_reg );
-  libxsmm_x86_instruction_pop_reg( io_generated_code, io_gp_reg );
```
Comment (alheinecke) [inline src/generator_common_x86.c:1971]: Why is touching this routine needed this is also used in GEMM... is this because originally we had only 4 pops?

**Review by alheinecke — CHANGES_REQUESTED** [review]

File: src/generator_common_x86.c (hunk 5 above)
Comment (nkmellem) [reply src/generator_common_x86.c:1971]: yes, it originally had only 4 POPs, i was trying to reuse this code which introduced that bug causing BF16 kernels to fail. its restored now.

File: samples/eltwise/run_eltwise_avx512.sh (hunk 1 above)
Comment (nkmellem) [reply samples/eltwise/run_eltwise_avx512.sh:1]: Done.

File: samples/eltwise/eltwise_common.h (hunk 2 above)
Comment (nkmellem) [reply samples/eltwise/eltwise_common.h:16]: Done.

File: include/libxsmm_typedefs.h (hunk 3 above)
Comment (nkmellem) [reply include/libxsmm_typedefs.h:340]: done.

**Review by alheinecke — APPROVED** [review]

**Pull request merged**